lib/fastlane/plugin/react_native_release/actions/react_native_release.rb

```diff
@@ -4,45 +4,39 @@
 module Fastlane
   module Actions
     class ReactNativeReleaseAction < Action
-      VALID_TARGETS = %w{beta production}
+      VALID_RELEASE_TYPES = %w{beta release hotfix}
 
+      # params:
+      # release_type, alpha_branch, beta_branch, release_branch, hotfix_destination
       def self.run(params)
         require 'fastlane/plugin/android_versioning'
-
         other_action.create_fastlane_session
 
-        target = UI.select "Select a release type:", VALID_TARGETS
-        is_beta = target.include?('beta')
-        is_hotfix = params[:hotfix] === true
+        release_type = params[:release_type] || UI.select("Select a release type:", VALID_RELEASE_TYPES)
+        is_beta = release_type.include?('beta')
```

Comment:
Not important at all, but these three lines (and the following conditional) could be done metaprogram-y  to set the state for more release types in the future.

Reply:
totally agree. I want to rework this whole action in a followup PR to properly support master-based and gitflow-based workflows (and hotfixes). I'll try and clean up there as well.